src/main/java/ay2021s1_cs2103_w16_3/finesse/storage/JsonAdaptedMonthlyBudget.java

```diff
@@ -0,0 +1,55 @@
+package ay2021s1_cs2103_w16_3.finesse.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import ay2021s1_cs2103_w16_3.finesse.commons.exceptions.IllegalValueException;
+import ay2021s1_cs2103_w16_3.finesse.model.budget.MonthlyBudget;
+import ay2021s1_cs2103_w16_3.finesse.model.budget.MonthlyExpenseLimit;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Expense;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Title;
+
+/**
+ * JSON-friendly version of {@link Expense}.
+ */
+class JsonAdaptedMonthlyBudget {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Monthly Budget's %s field is missing!";
+
+    private final String expenseLimit;
+
+    /**
+     * Constructs a {@code JsonAdaptedExpense} with the given transaction details.
+     */
+    @JsonCreator
+    public JsonAdaptedMonthlyBudget(@JsonProperty("expenseLimit") String expenseLimit) {
+        this.expenseLimit = expenseLimit;
+    }
+
+    /**
+     * Converts a given {@code Expense} into this class for Jackson use.
+     */
+    public JsonAdaptedMonthlyBudget(MonthlyBudget source) {
+        expenseLimit = source.getMonthlyExpenseLimit().getObservableValue().get().toString();
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted expense object into the model's {@code Expense} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted transaction.
+     */
+    public MonthlyBudget toModelType() throws IllegalValueException {
+        final MonthlyBudget monthlyBudget = new MonthlyBudget();
+        if (expenseLimit == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Title.class.getSimpleName()));
+        }
+
+        if (!MonthlyExpenseLimit.isValidAmount(expenseLimit)) {
```

Comment:
Since the user was allowed to input an amount of more than 2 decimal places. When I restarted the application the application opened up with no data on the previously stored transactions. An exception was thrown and then an empty FinanceTracker was started. The image below shows the exception being thrown when I opened the application again:
<img width="988" alt="Screenshot 2020-10-23 at 11 00 36 AM" src="https://user-images.githubusercontent.com/48427064/96951399-ffd0ac80-151e-11eb-98c4-0639ce135973.png">

Reply:
This is also due to the fact that when amounts are being written, it is using `BigDecimal::toString()`, where perhaps better abstractions could be done.

![image](https://user-images.githubusercontent.com/53135010/96952036-6d310d00-1520-11eb-9c85-0916f05e3d0f.png)